modules/dm_Recycler/clients/base/views/record/record.js

```diff
@@ -1,9 +1,73 @@
 ({
     extendsFrom: 'RecordView',
 
+ //    events: {
+	// 	'click *' : 'preventClick'
+	// },
+
     initialize: function(options){
         console.log("dm_Recycler Record View Initialization.");
         this._super("initialize",[options]);
+    },
+
+    delegateButtonEvents: function() {
+        this._super("delegateButtonEvents");
+        this.context.on('button:restore_button:click', this.restoreRecord, this);
+    },
+
+    restoreRecord: function(){
+    	
+    	var self = this;
+
+    	//Disable buttons to avoide collision clicks
+    	this.toggleButtons(false);
+
+    	//Add loading message...
+    	app.alert.show('Recycler_restoreRecord', {
+            level: 'process'
+        });
+
+    	var url = app.api.buildURL('Recycler', this.model.id);
+
+		var params = {
+            bean_module: this.model.get("bean_module"),
+            bean_id: this.model.get("bean_id")
```

Comment:
I still think, we should jsut pass something like "restore" as a parameter. The module and id are already there, so if I just instantiate a Recycler Bean, I could then just call dm_Recycler->restore and have all Logic for setting restored field on bean, and restoring actual record in one easy to use spot.